src/tools/unpack.c

```diff
@@ -937,32 +939,38 @@ void unpack_cleanup(void *user_data)
     for (int i=0; i<opts->n_sourcedirs; i++) {
         FREE_VAR(opts->sourcedirs[i]);
     }
+
+    sfmf_control_close();
 }
 
 static void draw_progress(struct UnpackOptions *opts, int i, const char *message)
 {
     static float last_progress = -1.f;
 
-    if (opts->progress) {
-        float partial = (opts->header.entries_length && i >= 0)
-            ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
-            : (i == -1 ? 0.f : 1.f);
+    float partial = (opts->header.entries_length && i >= 0)
+        ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
```

Comment:
Really minor, but dropping unnecessary casts & parentheses could improve readability or at least make the equations shorter...

  fminf(1.f, i / (float)opts->header.entries_length)

and few lines below

  float progress = fminf(1.f, (opts->steps.current + partial) / (float)opts->steps.total);